parsers/robot.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * robot.c
+ * parser for robot framewor
+ * Author - Daniel Riechers
+ * License GPL-2
+ */
+
+#include "general.h"
+
+#include <string.h>
+
+#include "parse.h"
+#include "vstring.h"
+#include "routines.h"
+
+typedef enum {
+        K_KEYWORD,
+        K_VARIABLE,
+        COUNT_KIND
+} RobotKind;
+
+static kindOption RobotKinds[COUNT_KIND] = {
+	{true, 'k', "keyword",    "keywords"},
+	{true, 'v', "variable",   "variables"},
+};
+
+static void findRobotTags (void)
+{
+	findRegexTags ();
+}
+
+static void whitespaceSwap (vString *const s)
+{
+        char replaceWith = '_';
+        char toReplace = ' ';
+
+        if(strchr(s->buffer, '_'))
+        {
+            replaceWith = ' ';
+            toReplace = '_';
+        }
+
+        for(int i=0; i < s->length; i++)
+            if(s->buffer[i] == toReplace)
+                s->buffer[i] = replaceWith;
+}
+
+static void keywords (const char *const line, const regexMatch *const matches,
+                               const unsigned int count, void *data)
+{
+    if (count > 1)
+    {
+        vString *const name = vStringNew ();
+        vStringNCopyS (name, line + matches [1].start, matches [1].length);
+        makeSimpleTag (name, RobotKinds, K_KEYWORD);
+        whitespaceSwap(name);
+        makeSimpleTag (name, RobotKinds, K_KEYWORD);
```

Comment:
I would like to know whitespaceSwap more.
You call whitespaceSwap then call makeSimpleTag to make a new tag.

My understanding is that you convert the representation of keyword from robot to  python, a language used in writing library level keyword. Am I correct?